drivers/pinctrl/pinctrl_numicro.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022 SEAL AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_numicro_pinctrl
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/dt-bindings/pinctrl/numicro-pinctrl.h>
+#include <soc.h>
```

Comment:
why is this needed? please use `<NuMicro.h>` instead.

As a general rule, include necessary headers instead of relying on indirect includes, some I immediately see:

```
<stdint.h>

<zephyr/devicetree.h>
<zephyr/sys/util.h>
```